apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -418,28 +418,28 @@ suite('ActivityHandler', function() {
 
       test('email in notification title when contact without name', function() {
         sinon.assert.called(sendSpy);
-        var notification = sendSpy.firstCall.thisValue;
-        assert.equal(notification.title, emailAddress);
+        var notification = sendSpy.firstCall;
+        assert.isTrue(notification.calledWith(emailAddress));
       });
     });
 
     suite('after getSelf', function() {
       var sendSpy, contactSpy;
       setup(function(done) {
-        sendSpy = this.sinon.spy(window, 'Notification');
+        sendSpy = this.sinon.spy(NotificationHelper, 'send');
         contactSpy = this.sinon.spy(Contacts, 'findByAddress');
         MockNavigatormozApps.mTriggerLastRequestSuccess().then(done, done);
       });
 
       test('a notification is sent', function() {
         sinon.assert.called(sendSpy);
-        var notification = sendSpy.firstCall.thisValue;
-        assert.equal(notification.body, message.body);
-        assert.deepEqual(notification.data, {
+        var notification = sendSpy.firstCall;
+        assert.equal(notification.args[1].body, message.body);
+        assert.deepEqual(notification.args[1].data, {
           id: message.id,
           threadId: message.threadId
         });
-        assert.equal(notification.icon, 'sms');
+        assert.equal(notification.args[1].icon, 'sms');
```

Comment:
Hmm, tried this:

```
        sinon.assert.calledWithMatch(sendSpy, {
          body: message.body,
          data: {
            id: message.id,
            threadId: message.threadId
          },
          icon: 'sms',
          tag: "threadId:1"
        });
```

and the error is: 
```
  1) [sms-views/shared/test/unit/activity_handler_test.js] ActivityHandler sms received after getSelf a notification is sent:
     Error: expected send to be called with match { body: "body", data: { id: 1, threadId: 1 }, icon: "sms", tag: "threadId:1" }
    send(Pepito O'Hare, { body: "body", data: { id: 1, threadId: 1 }, icon: "sms", tag: "threadId:1" }) => { then: function () {} }
```

which is weird because it's exactly how it is supposed to work :/